USER_MANUAL.md

```diff
@@ -1,12 +1,355 @@
+
+We now present our high-level BNF syntax:
+
+```
+  File         ::= RequiresList ModuleList
+
+  RequiresList ::= "" | Requires RequiresList
+  Requires     ::= RequireCmd {String}
+  RequireCmd   ::= "requires" | "require"
+
+  Module       ::= "module" {ModuleId} AttributeSet ImportList SentenceList "endmodule"
+
+  AttributeSet ::= "" | "[" Attributes "]"
+  Attributes   ::= Attributes "," {Attribute}
+
+  ImportList   ::= "" | Import ImportList
+  Import       ::= ImportCmd {ModuleId}
+  ImportCmd    ::= "imports" | "import"
+
+  SentenceList ::= "" | Sentence SentenceList
+  Sentence     ::= SortDeclaration
+                 | GrammarDeclaration
+                 | LexicalDeclaration
+                 | AssociativityDeclaration
+                 | PriorityDeclaration
+                 | ConfigDeclaration
+                 | RuleDeclaration
+                 | ContextDeclaration
+                 | ContextAliasDeclaration
+```
+
+We can further drill down by sentence type.
+
+```
+  SortDeclaration    ::= "syntax" {SortId} AttributeSet
+
+  GrammarDeclaration ::= "syntax" {SortId} "::=" AlternativeList
+  AlternativeList    ::= Production
+                       | Production "|" AlternativeList
+
+  Production         ::= ProductionSyntax AttributeSet
+  ProductionSyntax   ::= FunctionSyntax
+                       | GenericSyntax
+
+  FunctionNotation   ::= {Id} "(" ParameterList ")"
+  ParameterList      ::= "" | NeParameterList
+  NeParameterList    ::= Parameter | Parameter "," NeParameterList
+  Parameter          ::= OptParamName {SortId}
+  OptParamName       ::= "" | {Id} ":"
+
+  GenericNotation    ::= StringOrSortIdList
+  StringOrSortIdList ::= StringOrSortId | StringOrSortId StringOrSortIdList
+  StringOrSortIdList ::= {String} | {SortId}
+
+  LexicalDeclaration ::= "syntax" "lexical" {Id} "=" {String}
+
+  AssociativityDeclaration ::= "syntax" Associativity NeIdList
+  NeIdList                 ::= {Id} | {Id} NeIdList
+  Associativity            ::= "left" | "right" | "non-assoc"
+
+  PriorityDeclaration ::= "syntax" "priorities" PriorityList
+  PriorityList        ::= NeIdList ">" PriorityList
+                        | NeIdList ">" NeIdList
+
+  ConfigDeclaration ::= "configuration" NeCellDeclList
+  NeCellDeclList    ::= CellDecl | CellDecl NeCellDeclList
+  CellDecl          ::= ConfigImport
+                      | StartCellDecl InnerCellDecl EndCellDecl
+  ConfigImport      ::= "<" {CellId} "/>"
+  StartCellDecl     ::= "<" {CellId} CellAttributeList ">"
+  InnerCellDecl     ::= {TermBubble} | NeCellDeclList
+  EndCellDecl       ::= "</" {CellId} ">"
+  CellAttributeList ::= "" | CellAttribute CellAttributeList
+  CellAttribute     ::= {Id} "=" {String}
+
+  RuleDeclaration   ::= "rule" OptName {BubbleWithRewrites} RequiresCondition EnsuresCondition AttributeSet
+  OptName           ::= "" | Name
+  Name              ::= "[" {Id} "]" ":"
+  RequiresCondition ::= "" | "requires" {Bubble}
```

Comment:
this is also not how bubbles in rules work

Reply:
I removed this section from this PR because it needs more polish.